agent/benchmark/observable-actors.rb

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env ruby
+
+require_relative './benchmark-support'
+
+class TestActor
+  include Celluloid
+  include Kontena::Observable::Helper
+
+  DELAY = getenv('START_DELAY', 1.0) { |x| Float(x) }
+
+  def reset
+    @ready = nil
+    observable.reset
+  end
+
+  def start(delay: DELAY)
+    sleep delay
+    @ready = Time.now
+    observable.update @ready
+  end
+
+  def ready?
+    @ready
+  end
+end
+
+class TestObserverActor
+  include Celluloid
+  include Kontena::Observer::Helper
+
+  def initialize(observable)
+    @observable = observable
+  end
+
+  def watch
+    ready = observe(@observable)
+    Time.now - ready
+  end
+end
+
+class TestWaitActor
+  include Celluloid
+  include Kontena::Helpers::WaitHelper
+
+  WAIT_INTERVAL = getenv("WAIT_INTERVAL", 0.01) { |v| Float(v) }
+
+  def initialize(actor)
+    @actor = actor
+  end
+
+  def watch
+    ready = wait_until!("observable ready", interval: WAIT_INTERVAL) { @actor.ready? }
+    Time.now - ready
+  end
+end
+
+supervise_registry = Kontena::Observable::Registry.supervise :as => :observable_registry
+test_actor = nil
+test_observable = nil
+
+benchmark(
+  {
+    'wait'      => ->(id) { TestWaitActor.new(test_actor).future.watch },
+    'observer'  => ->(id) { TestObserverActor.new(test_observable).future.watch },
```

Comment:
These results are invalid... the benchmark time includes the actor setup, and the results depend heavily on the ordering:

```
               user     system      total        real
wait       2.120000   0.290000   2.410000 (  3.031371)
observer   2.660000   0.150000   2.810000 (  3.697485)
                    delay
wait             0.814029
observer         0.900439
```

```
               user     system      total        real
observer   1.520000   0.150000   1.670000 (  2.545030)
wait       3.310000   0.280000   3.590000 (  4.184796)
                    delay
observer         0.431958
wait             1.279973
```

```

```               user     system      total        real
wait       2.160000   0.290000   2.450000 (  3.057220)
                    delay
wait             0.821810
```
```
               user     system      total        real
observer   1.510000   0.100000   1.610000 (  2.505361)
                    delay
observer         0.406092
```
```